dts/bindings/clock-management/clock-source.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Generic clock source that may be gated. This node accepts one specifier,
+  which either gates the clock (when set to 0), or ungates the clock
+  (when nonzero)
+  Other specifier values may cause undefined behavior.
+
+compatible: "clock-source"
```

Comment:
I don't see where this compat is used anywhere, what is the purpose of this?
There is a gate-offset in the source? Should it be a combination of div, mux, gate and output?

I see there is a [nxp,syscon-clock-gate.yaml](https://github.com/zephyrproject-rtos/zephyr/pull/72102/files#diff-6a61e1650cc8385d19abe4ae882ed5b5edd041ae942f9bdc07231246b61cb0fc), but no base binding for clock gate?